### PR TITLE
Add FriendRoomMessageSelectPage

### PR DESCRIPTION
**Pull request by vabold**
Aims to solve `Send chat messages` in #382.

### CODE REVIEW

File: payload/game/ui/FriendRoomMessageSelectPage.cc [hunk 1]
```diff
@@ -0,0 +1,178 @@
+#include "FriendRoomMessageSelectPage.hh"
+
+#include "game/ui/Font.hh"
+#include "game/ui/MenuInputManager.hh"
+#include "sp/cs/RoomClient.hh"
```
Comment (stblr) [inline payload/game/ui/FriendRoomMessageSelectPage.cc:5]: Brackets

File: payload/game/ui/FriendRoomMessageSelectPage.cc [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    m_messageSelects[0].load();
+    m_messageSelects[1].load();
+    m_sheetSelect.load("button", "CommentSelectArrowRight", "ButtonArrowRight", "CommentSelectArrowLeft", "ButtonArrowLeft", 1, false, false);
+    m_messageSelectPageNum.load("control", "MessageSelectPageNum", "MessageSelectPageNum", nullptr);
+    m_friendRoomMessageSelectObiBottom.load("bg", "FriendRoomMessageSelectObiBottom", "MenuObyBottom", nullptr);
+    m_backButton.load("button", "FriendRoomMessageSelectBack", "ButtonBack", 1, false, true);
+
+    m_sheetSelect.setRightHandler(&m_onRight);
+    m_sheetSelect.setLeftHandler(&m_onLeft);
+    m_backButton.setFrontHandler(nullptr, false);
```
Comment (stblr) [inline payload/game/ui/FriendRoomMessageSelectPage.cc:37]: Missing handler

**Review by stblr — CHANGES_REQUESTED** [review]

File: payload/game/ui/FriendRoomMessageSelectPage.cc (hunk 1 above)
Comment (vabold) [reply payload/game/ui/FriendRoomMessageSelectPage.cc]: Resolved in https://github.com/stblr/mkw-sp/pull/392/commits/b289e5a98477d62154a17ca746a3689fe9a865f8.

File: payload/game/ui/FriendRoomMessageSelectPage.cc [hunk 3]
```diff
@@ -0,0 +1,178 @@
+#include "FriendRoomMessageSelectPage.hh"
+
+#include "game/ui/Font.hh"
+#include "game/ui/MenuInputManager.hh"
```
[the remark this answers is not in the document]
Comment (vabold) [reply payload/game/ui/FriendRoomMessageSelectPage.cc]: Resolved in https://github.com/stblr/mkw-sp/pull/392/commits/b289e5a98477d62154a17ca746a3689fe9a865f8.

File: payload/game/ui/FriendRoomMessageSelectPage.cc (hunk 2 above)
Comment (vabold) [reply payload/game/ui/FriendRoomMessageSelectPage.cc]: Resolved in https://github.com/stblr/mkw-sp/pull/392/commits/b289e5a98477d62154a17ca746a3689fe9a865f8.

File: payload/game/ui/FriendRoomMessageSelectPage.hh [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        Register = 2,
+    };
+
+    FriendRoomMessageSelectPage();
+    ~FriendRoomMessageSelectPage() override;
+
+    void onInit() override;
+    void onActivate() override;
+    void onDeactivate() override;
+    void initText();
```
[the remark this answers is not in the document]
Comment (vabold) [reply payload/game/ui/FriendRoomMessageSelectPage.hh]: Resolved in https://github.com/stblr/mkw-sp/pull/392/commits/b289e5a98477d62154a17ca746a3689fe9a865f8.

File: payload/game/ui/FriendRoomMessageSelectPage.hh [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    SheetSelectControlScaleFade m_sheetSelect;
+    LayoutUIControlScaleFade m_messageSelectPageNum;
+    LayoutUIControl m_friendRoomMessageSelectObiBottom;
+    CtrlMenuBackButton m_backButton;
+    MessageSelectControl *m_visibleMessageSelect;
+    MessageSelectControl *m_hiddenMessageSelect;
+    MenuType m_menuType;
+    s32 m_messageCount;
+    s32 m_pageCount;
+    s32 m_currentPageIdx;
```
[the remark this answers is not in the document]
Comment (vabold) [reply payload/game/ui/FriendRoomMessageSelectPage.hh]: Resolved in https://github.com/stblr/mkw-sp/pull/392/commits/b289e5a98477d62154a17ca746a3689fe9a865f8.

**Review by stblr — APPROVED** [review]

**Pull request merged**